src/pages/ScholarshipDetails.jsx

```diff
@@ -16,27 +16,30 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
-export default function ScholarshipDetailsPage({ match }) {
+export default function ScholarshipDetails({ history, location, match }) {
+  const classes = useStyles();
   const { id } = match.params;
-  const [scholarship, setScholarship] = useState();
-  const [loading, setLoading] = useState(true);
+  const [scholarship, setScholarship] = useState(location.state?.scholarship);
   const [error, setError] = useState();
-  const classes = useStyles();
+  const loading = !error && !scholarship;
 
-  useEffect(
-    () =>
-      Scholarships.id(id).subscribe(
-        (s) => {
-          document.title = `${BRAND_NAME} | ${s.data.name}`;
-          setScholarship(s);
-          setLoading(false);
-        },
-        (err) => {
-          setError(err);
-        }
-      ),
-    [id]
-  );
+  if (scholarship) {
+    document.title = `${BRAND_NAME} | ${scholarship.data.name}`;
+  }
+
+  // Clear location state in case of refresh/navigation to other pages.
+  useEffect(() => {
+    if (location.state.scholarship) {
+      history.replace({ state: {} });
+    }
+  }, [history, location]);
```

Comment:
This runs the first time the ScholarshipsDetails page is loaded, and then each time the `history` and `location` change?

Reply:
That's right. We don't have a good way to detect a page refresh so the best we can do at the moment is replace the state in history so that the next time this page is loaded it's no longer using it. Unfortunately this also means if you navigate away and come back then it refreshes as well. There might be a better solution using something else besides location state though (I think I saw React Redux state mentioned as a potential solution).